internal/tiger/config/credentials.go

```diff
@@ -0,0 +1,204 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/zalando/go-keyring"
+)
+
+// Keyring parameters
+const (
+	keyringServiceName = "tiger-cli"
+	keyringUsername    = "credentials"
+)
+
+// storedCredentials represents the JSON structure for stored credentials
+type storedCredentials struct {
+	APIKey    string `json:"api_key"`
+	ProjectID string `json:"project_id"`
+}
+
+// testServiceNameOverride allows tests to override the service name for isolation
+var testServiceNameOverride string
+
+// GetServiceName returns the appropriate service name for keyring operations
+func GetServiceName() string {
+	// Tests should set a unique service name to avoid conflicts
+	if testServiceNameOverride != "" {
+		return testServiceNameOverride
+	}
+
+	// In test mode without an override, panic to catch missing test setup
+	if testing.Testing() {
+		panic("test must call SetTestServiceName() to set a unique keyring service name")
+	}
+
+	return keyringServiceName
+}
+
+// SetTestServiceName sets a unique service name for testing based on the test name
+// This allows tests to use unique service names to avoid conflicts when running in parallel
+// The cleanup is automatically registered with t.Cleanup()
+func SetTestServiceName(t *testing.T) {
```

Comment:
I feel like this could be in the test file. Lots of test related code leaking into the implementation looks off to me, but maybe this is a common go pattern 🤷